<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Zenith gene set testing after dream analysis â€¢ zenith</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Zenith gene set testing after dream analysis">
<meta property="og:description" content="zenith">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">zenith</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.6</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/geuvadis.html">Zenith gene set testing after dream analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/GabrielHoffman/zenith/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="geuvadis_files/header-attrs-2.11/header-attrs.js"></script><script src="geuvadis_files/kePrint-0.0.1/kePrint.js"></script><link href="geuvadis_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Zenith gene set testing after dream analysis</h1>
            <h3 class="subtitle">Zenith controls the false positive rate while retaining power</h3>
                        <h4 class="author"><a href="http://gabrielhoffman.github.io">Gabriel Hoffman</a></h4>
            <address class="author_afil">
      <p>Icahn School of Medicine at Mount Sinai, New York</p>
<br><h4 class="date"></h4>
<p>Thu Nov 4 22:07:24 2021</p>
      
      <small class="dont-index">Source: <a href="https://github.com/GabrielHoffman/zenith/blob/master/vignettes/geuvadis.Rmd"><code>vignettes/geuvadis.Rmd</code></a></small>
      <div class="hidden name"><code>geuvadis.Rmd</code></div>

    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Perform gene set analysis on the result of differential expression using linear (mixed) modeling with <a href="https://doi.org/10.1093/bioinformatics/btaa687">dream</a> by considering the correlation between gene expression traits. This package implements the <a href="https://www.rdocumentation.org/packages/limma/versions/3.28.14/topics/camera">camera</a> method from the <a href="https://bioconductor.org/packages/limma/">limma</a> package proposed by <a href="https://doi.org/10.1093/nar/gks461">Wu and Smyth (2012)</a>. <code><a href="../reference/zenith.html">zenith()</a></code> is a simple extension of <code><a href="https://rdrr.io/pkg/limma/man/camera.html">camera()</a></code> to be compatible with linear (mixed) models implemented in <code><a href="https://rdrr.io/pkg/variancePartition/man/dream-method.html">dream()</a></code>.</p>
    </div>
    
<div id="load-and-process-rna-seq-data" class="section level2">
<h2 class="hasAnchor">
<a href="#load-and-process-rna-seq-data" class="anchor"></a>Load and process RNA-seq data</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/leekgroup/recount">recount</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/variancePartition">variancePartition</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma">limma</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">zenith</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">EnrichmentBrowser</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/">kableExtra</a></span><span class="op">)</span>

<span class="co"># For ENSEMBL id ENSG00000279457.4, return ENSG00000279457</span>
<span class="va">trim_ensembl_ids</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"(.*)\\.(.*)"</span>, <span class="st">"\\1"</span>, <span class="va">x</span><span class="op">)</span> 
<span class="op">}</span></code></pre></div>
<div id="load-rna-seq-data-and-metadata" class="section level4">
<h4 class="hasAnchor">
<a href="#load-rna-seq-data-and-metadata" class="anchor"></a>Load RNA-seq data and metadata</h4>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Download GEUVADIS data from recount2</span>
<span class="kw">if</span><span class="op">(</span> <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/files.html">file.exists</a></span><span class="op">(</span><span class="st">"ERP001942/rse_gene.Rdata"</span><span class="op">)</span> <span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html">download_study</a></span><span class="op">(</span><span class="st">'ERP001942'</span><span class="op">)</span>
<span class="op">}</span>

<span class="co"># Load the data</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span><span class="op">(</span><span class="st">'ERP001942'</span>, <span class="st">'rse_gene.Rdata'</span><span class="op">)</span><span class="op">)</span>

<span class="co"># scale counts</span>
<span class="va">rse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html">scale_counts</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span>

<span class="co"># get count data</span>
<span class="va">geneCounts</span> <span class="op">=</span> <span class="fu">assay</span><span class="op">(</span><span class="va">rse</span><span class="op">)</span>

<span class="co"># read and format metadata</span>
<span class="va">info</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html">read.table</a></span><span class="op">(</span><span class="st">"https://www.ebi.ac.uk/arrayexpress/files/E-GEUV-1/E-GEUV-1.sdrf.txt"</span>, header<span class="op">=</span><span class="cn">TRUE</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"Comment\\."</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"Characteristics\\."</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"\\.$"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span><span class="op">)</span>

<span class="co"># extract unique samples and drop a problematic sample</span>
<span class="va">info</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">info</span><span class="op">[</span>,<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span> <span class="op">%in%</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">'Scan.Name'</span>, <span class="st">'SUBMITTED_FILE_NAME'</span>, <span class="st">'FASTQ_URI'</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="va">info</span> <span class="op">=</span> <span class="va">info</span><span class="op">[</span><span class="va">info</span><span class="op">$</span><span class="va">ENA_RUN</span> <span class="op">!=</span> <span class="st">'ERR188328'</span>,<span class="op">]</span>

<span class="co"># merge sample information in colData(rse) and info</span>
<span class="va">metadata</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html">merge</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">rse</span><span class="op">)</span>, <span class="va">info</span>, by.x<span class="op">=</span><span class="st">"run"</span>, by.y<span class="op">=</span><span class="st">'ENA_RUN'</span><span class="op">)</span>
<span class="va">metadata</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span> <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">run</span>

<span class="co"># keep RNA-seq samples in metadata</span>
<span class="va">geneCounts</span> <span class="op">=</span> <span class="va">geneCounts</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span> <span class="op">%in%</span> <span class="va">metadata</span><span class="op">$</span><span class="va">run</span><span class="op">]</span>
<span class="va">geneCounts</span> <span class="op">=</span> <span class="va">geneCounts</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">$</span><span class="va">run</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></code></pre></div>
</div>
<div id="rna-seq-processing" class="section level4">
<h4 class="hasAnchor">
<a href="#rna-seq-processing" class="anchor"></a>RNA-seq processing</h4>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># keep only expressed genes</span>
<span class="va">isexpr</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colSums.html">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html">cpm</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">1</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">0.2</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">ncol</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span>

<span class="co"># apply TMM normalization</span>
<span class="va">dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span><span class="op">(</span> <span class="va">geneCounts</span><span class="op">[</span><span class="va">isexpr</span>,<span class="op">]</span> <span class="op">)</span>
<span class="va">dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span><span class="op">(</span><span class="va">dge</span><span class="op">)</span>

<span class="co"># Create log2 counts per million for plotting</span>
<span class="va">geneExpr</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html">cpm</a></span><span class="op">(</span><span class="va">dge</span>, log<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># extract gene annotations</span>
<span class="va">geneInfo</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">rse</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div id="confirm-sex-based-on-expression" class="section level4">
<h4 class="hasAnchor">
<a href="#confirm-sex-based-on-expression" class="anchor"></a>Confirm sex based on expression</h4>
<p>Plot expression of XIST and UTY<br>
We see a clear separation of sex based on XIST from chrX and UTY from chrY.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># extract ENSEMBL identifies for XIST and UTY</span>
<span class="va">ens_XIST</span> <span class="op">=</span> <span class="va">geneInfo</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">geneInfo</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"XIST"</span><span class="op">)</span><span class="op">]</span>
<span class="va">ens_UTY</span> <span class="op">=</span> <span class="va">geneInfo</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span><span class="op">(</span><span class="va">geneInfo</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"UTY"</span><span class="op">)</span><span class="op">]</span>

<span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="va">metadata</span>, XIST<span class="op">=</span> <span class="va">geneExpr</span><span class="op">[</span><span class="va">ens_XIST</span>,<span class="op">]</span>,
        UTY <span class="op">=</span> <span class="va">geneExpr</span><span class="op">[</span><span class="va">ens_UTY</span>,<span class="op">]</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span><span class="va">XIST</span>, <span class="va">UTY</span>, color<span class="op">=</span><span class="va">sex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"blue"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="geuvadis_files/figure-html/plot.sex-1.png" width="480"></p>
</div>
<div id="differential-expression-with-dream" class="section level4">
<h4 class="hasAnchor">
<a href="#differential-expression-with-dream" class="anchor"></a>Differential expression with <code>dream</code>
</h4>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">form</span> <span class="op">=</span> <span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">ancestry.category</span> <span class="op">+</span> <span class="va">laboratory</span>
<span class="va">vobj</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/variancePartition/man/voomWithDreamWeights.html">voomWithDreamWeights</a></span><span class="op">(</span><span class="va">dge</span>, <span class="va">form</span>, <span class="va">metadata</span>, plot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="geuvadis_files/figure-html/diffExpr-1.png" width="480"></p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/variancePartition/man/dream-method.html">dream</a></span><span class="op">(</span><span class="va">vobj</span>, <span class="va">form</span>, <span class="va">metadata</span>, computeResiduals<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></code></pre></div>
</div>
<div id="load-genesets" class="section level4">
<h4 class="hasAnchor">
<a href="#load-genesets" class="anchor"></a>Load genesets</h4>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load Gene Ontology database </span>
<span class="va">go.gs</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_GeneOntology.html">get_GeneOntology</a></span><span class="op">(</span><span class="op">)</span>
   
<span class="co"># convert from GeneSetCollection to list used by camera and zenith</span>
<span class="va">geneSets_GO</span> <span class="op">=</span> <span class="fu"><a href="../reference/recodeToList.html">recodeToList</a></span><span class="op">(</span> <span class="va">go.gs</span> <span class="op">)</span></code></pre></div>
</div>
</div>
<div id="zenith-gene-set-analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#zenith-gene-set-analysis" class="anchor"></a>Zenith gene set analysis</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Map from Ensembl genes in geneSets_GO to </span>
<span class="co"># from trimmed Ensembl names from RNA-seq data </span>
<span class="va">index</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ids2indices.html">ids2indices</a></span><span class="op">(</span> <span class="va">geneSets_GO</span>, <span class="fu">trim_ensembl_ids</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
   
<span class="co"># filter by size of gene set</span>
<span class="va">n_genes_in</span> <span class="op">=</span> <span class="fl">10</span>
<span class="va">index</span> <span class="op">=</span> <span class="va">index</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">index</span>, <span class="va">length</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">n_genes_in</span><span class="op">]</span>

<span class="co"># Run camera</span>
<span class="va">dsgn</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span> <span class="va">form</span>, <span class="va">metadata</span><span class="op">)</span>
<span class="va">res_camera</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/camera.html">camera</a></span><span class="op">(</span> <span class="va">vobj</span>, <span class="va">index</span>, <span class="va">dsgn</span>, inter.gene.cor<span class="op">=</span><span class="cn">NA</span>, contrast<span class="op">=</span><span class="st">"sexmale"</span><span class="op">)</span>

<span class="co"># Run zenith.  </span>
<span class="co"># Note that zenith allows analysis of any dream model and </span>
<span class="co"># any coefficient or contrast defined in the model. </span>
<span class="co"># Setting inter.gene.cor=NA estimates the correlation from the model</span>
<span class="va">res_zenith</span> <span class="op">=</span> <span class="fu"><a href="../reference/zenith.html">zenith</a></span><span class="op">(</span> <span class="va">fit</span>, <span class="st">"sexmale"</span>, <span class="va">index</span>, inter.gene.cor<span class="op">=</span><span class="cn">NA</span><span class="op">)</span></code></pre></div>
<div id="compare-camera-and-zenith" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-camera-and-zenith" class="anchor"></a>Compare camera and zenith</h3>
<div id="results-from-camera" class="section level4">
<h4 class="hasAnchor">
<a href="#results-from-camera" class="anchor"></a>Results from camera</h4>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_camera</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>,<span class="op">]</span> <span class="op">%&gt;%</span> <span class="va">kable</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>full_width <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </code></pre></div>
<table class="table table">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:right;">
Correlation
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:right;">
FDR
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GO0060765: regulation of androgen receptor signaling pathway
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
-0.0126289
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:right;">
0.0045249
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0050681: androgen receptor binding
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
-0.0049552
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000263
</td>
<td style="text-align:right;">
0.0713334
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0030521: androgen receptor signaling pathway
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
0.0073531
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000350
</td>
<td style="text-align:right;">
0.0713334
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0045495: pole plasm
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.0066687
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000638
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0001941: postsynaptic membrane organization
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
-0.0006421
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000702
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0003785: actin monomer binding
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
-0.0068675
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000971
</td>
<td style="text-align:right;">
0.0988046
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0097110: scaffold protein binding
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
0.0032819
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0001190
</td>
<td style="text-align:right;">
0.1038613
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0035176: social behavior
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.0496503
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0016764
</td>
<td style="text-align:right;">
0.9997348
</td>
</tr>
</tbody>
</table>
</div>
<div id="results-from-zenith" class="section level4">
<h4 class="hasAnchor">
<a href="#results-from-zenith" class="anchor"></a>Results from zenith</h4>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_zenith</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>,<span class="op">]</span> <span class="op">%&gt;%</span> <span class="va">kable</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>full_width <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>     </code></pre></div>
<table class="table table">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:right;">
Correlation
</th>
<th style="text-align:right;">
delta
</th>
<th style="text-align:right;">
se
</th>
<th style="text-align:right;">
p.less
</th>
<th style="text-align:right;">
p.greater
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
FDR
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GO0060765: regulation of androgen receptor signaling pathway
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
-0.0138619
</td>
<td style="text-align:right;">
2.394862
</td>
<td style="text-align:right;">
0.4769345
</td>
<td style="text-align:right;">
0.9999996
</td>
<td style="text-align:right;">
0.0000004
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0045249
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0030521: androgen receptor signaling pathway
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
0.0058599
</td>
<td style="text-align:right;">
1.793398
</td>
<td style="text-align:right;">
0.4213249
</td>
<td style="text-align:right;">
0.9999874
</td>
<td style="text-align:right;">
0.0000126
</td>
<td style="text-align:right;">
0.0000253
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0535272
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0050681: androgen receptor binding
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
-0.0055522
</td>
<td style="text-align:right;">
1.777379
</td>
<td style="text-align:right;">
0.4184603
</td>
<td style="text-align:right;">
0.9999869
</td>
<td style="text-align:right;">
0.0000131
</td>
<td style="text-align:right;">
0.0000263
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0535272
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0045495: pole plasm
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.0036462
</td>
<td style="text-align:right;">
2.722434
</td>
<td style="text-align:right;">
0.6744655
</td>
<td style="text-align:right;">
0.9999681
</td>
<td style="text-align:right;">
0.0000319
</td>
<td style="text-align:right;">
0.0000638
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0001941: postsynaptic membrane organization
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
-0.0019566
</td>
<td style="text-align:right;">
1.964205
</td>
<td style="text-align:right;">
0.4894403
</td>
<td style="text-align:right;">
0.9999649
</td>
<td style="text-align:right;">
0.0000351
</td>
<td style="text-align:right;">
0.0000702
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0003785: actin monomer binding
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
-0.0087844
</td>
<td style="text-align:right;">
2.035066
</td>
<td style="text-align:right;">
0.5174099
</td>
<td style="text-align:right;">
0.9999515
</td>
<td style="text-align:right;">
0.0000485
</td>
<td style="text-align:right;">
0.0000971
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0911182
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0097110: scaffold protein binding
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
0.0027498
</td>
<td style="text-align:right;">
1.458051
</td>
<td style="text-align:right;">
0.3724309
</td>
<td style="text-align:right;">
0.9999478
</td>
<td style="text-align:right;">
0.0000522
</td>
<td style="text-align:right;">
0.0001044
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0911182
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0035176: social behavior
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.0479135
</td>
<td style="text-align:right;">
1.996151
</td>
<td style="text-align:right;">
0.6253904
</td>
<td style="text-align:right;">
0.9992437
</td>
<td style="text-align:right;">
0.0007563
</td>
<td style="text-align:right;">
0.0015127
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.9997350
</td>
</tr>
</tbody>
</table>
<p>While there is a very slight difference in the results from the methods, this is attributable to the very slight difference in the correlation value computed. This appears to be due to an numerical issue with <code>camera</code>, but values are correct to two digits.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html">merge</a></span><span class="op">(</span><span class="va">res_camera</span>, <span class="va">res_zenith</span>, by<span class="op">=</span><span class="st">"row.names"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html">plot</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">Correlation.x</span>, <span class="va">df</span><span class="op">$</span><span class="va">Correlation.y</span>, xlab<span class="op">=</span><span class="st">"Correlation from zenith"</span>, ylab<span class="op">=</span><span class="st">"Correlation from camera"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, col<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></code></pre></div>
<p><img src="geuvadis_files/figure-html/compare-1.png" width="480"></p>
</div>
</div>
</div>
<div id="analysis-with-msigdb" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis-with-msigdb" class="anchor"></a>Analysis with MSigDB</h2>
<div id="load-msigdb-and-perform-zenith-analysis" class="section level4">
<h4 class="hasAnchor">
<a href="#load-msigdb-and-perform-zenith-analysis" class="anchor"></a>Load <a href="https://www.gsea-msigdb.org/gsea/msigdb/index.jsp">MSigDB</a> and perform zenith analysis</h4>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gs_msigdb</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_MSigDB.html">get_MSigDB</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Map from Ensembl genes in gs_msigdb to </span>
<span class="co"># from trimmed Ensembl names from RNA-seq data </span>
<span class="va">index</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ids2indices.html">ids2indices</a></span><span class="op">(</span> <span class="fu"><a href="../reference/recodeToList.html">recodeToList</a></span><span class="op">(</span><span class="va">gs_msigdb</span><span class="op">)</span>, <span class="fu">trim_ensembl_ids</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># filter by size of gene set</span>
<span class="va">n_genes_in</span> <span class="op">=</span> <span class="fl">10</span>
<span class="va">index</span> <span class="op">=</span> <span class="va">index</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span><span class="op">(</span><span class="va">index</span>, <span class="va">length</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">n_genes_in</span><span class="op">]</span>

 <span class="co"># zenith analysis of MSigDB using square test statistics</span>
<span class="va">res_zenith_msigdb</span> <span class="op">=</span> <span class="fu"><a href="../reference/zenith.html">zenith</a></span><span class="op">(</span> <span class="va">fit</span>, <span class="st">"sexmale"</span>, <span class="va">index</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_zenith_msigdb</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span>,<span class="op">]</span> <span class="op">%&gt;%</span> <span class="va">kable</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html">kable_styling</a></span><span class="op">(</span>full_width <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>  </code></pre></div>
<table class="table table">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:right;">
Correlation
</th>
<th style="text-align:right;">
delta
</th>
<th style="text-align:right;">
se
</th>
<th style="text-align:right;">
p.less
</th>
<th style="text-align:right;">
p.greater
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
FDR
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
M5126_PYEON_CANCER_HEAD_AND_NECK_VS_CERVICAL_DN
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
19.224970
</td>
<td style="text-align:right;">
0.6067280
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M16091_DISTECHE_ESCAPED_FROM_X_INACTIVATION
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-14.671145
</td>
<td style="text-align:right;">
0.6150034
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M39089_ZHONG_PFC_C2_SOX5_BCL11B_POS_EXCITATORY_NEURON
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
6.648676
</td>
<td style="text-align:right;">
0.5695400
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M7727_RICKMAN_HEAD_AND_NECK_CANCER_B
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
4.895289
</td>
<td style="text-align:right;">
0.5303168
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M34944_HP_OLIGOSPERMIA
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
5.544998
</td>
<td style="text-align:right;">
0.6046005
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M22112_chrXp22
</td>
<td style="text-align:right;">
74
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-2.886132
</td>
<td style="text-align:right;">
0.3252910
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M22115_chrXq13
</td>
<td style="text-align:right;">
45
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-3.048338
</td>
<td style="text-align:right;">
0.3809299
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M37972_HP_NON_OBSTRUCTIVE_AZOOSPERMIA
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
4.306009
</td>
<td style="text-align:right;">
0.5563521
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M6596_GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_DN
</td>
<td style="text-align:right;">
119
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
2.193358
</td>
<td style="text-align:right;">
0.2881369
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M11031_CHEOK_RESPONSE_TO_MERCAPTOPURINE_AND_LD_MTX_UP
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
5.084481
</td>
<td style="text-align:right;">
0.7033105
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M32591_MIR151A_5P_MIR151B
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
4.783951
</td>
<td style="text-align:right;">
0.7034449
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
1.00e-07
</td>
</tr>
<tr>
<td style="text-align:left;">
M2778_TAVAZOIE_METASTASIS
</td>
<td style="text-align:right;">
54
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
2.370876
</td>
<td style="text-align:right;">
0.3587496
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
2.00e-07
</td>
</tr>
<tr>
<td style="text-align:left;">
M37270_HP_ABNORMAL_SPERMATOGENESIS
</td>
<td style="text-align:right;">
72
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
2.072099
</td>
<td style="text-align:right;">
0.3285087
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
1.30e-06
</td>
</tr>
<tr>
<td style="text-align:left;">
M6669_HEDENFALK_BREAST_CANCER_BRACX_UP
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-3.256621
</td>
<td style="text-align:right;">
0.5568357
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.66e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M22110_chrXp11
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-1.709990
</td>
<td style="text-align:right;">
0.2979060
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
2.81e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M17751_chr19p13
</td>
<td style="text-align:right;">
425
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-1.353669
</td>
<td style="text-align:right;">
0.2362841
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
2.81e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M8555_MODULE_489
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
3.819485
</td>
<td style="text-align:right;">
0.6740911
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
3.74e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M6593_GSE5099_DAY3_VS_DAY7_MCSF_TREATED_MACROPHAGE_DN
</td>
<td style="text-align:right;">
120
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
1.610435
</td>
<td style="text-align:right;">
0.2880756
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
5.32e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M561_chr19q13
</td>
<td style="text-align:right;">
527
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-1.280877
</td>
<td style="text-align:right;">
0.2318304
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1e-07
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
7.15e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M5384_GSE3982_MEMORY_CD4_TCELL_VS_BCELL_UP
</td>
<td style="text-align:right;">
143
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
1.522163
</td>
<td style="text-align:right;">
0.2774059
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1e-07
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
8.31e-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Gabriel Hoffman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
