<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Zenith gene set testing after dream analysis â€¢ zenith</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Zenith gene set testing after dream analysis" />
<meta property="og:description" content="zenith" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">zenith</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/geuvadis.html">Zenith gene set testing after dream analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/GabrielHoffman/zenith/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<script src="geuvadis_files/header-attrs-2.11/header-attrs.js"></script>
<script src="geuvadis_files/kePrint-0.0.1/kePrint.js"></script>
<link href="geuvadis_files/lightable-0.0.1/lightable.css" rel="stylesheet" />

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Zenith gene set testing after dream analysis</h1>
            <h3 class="subtitle">Zenith controls the false positive rate while retaining power</h3>
                        <h4 class="author"><a href="http://gabrielhoffman.github.io">Gabriel Hoffman</a></h4>
            <address class="author_afil">
      <p>Icahn School of Medicine at Mount Sinai, New York</p><br>                  
            <h4 class="date"><p>Mon Oct 18 09:59:59 2021</p></h4>
      
      <small class="dont-index">Source: <a href='https://github.com/GabrielHoffman/zenith/blob/master/vignettes/geuvadis.Rmd'><code>vignettes/geuvadis.Rmd</code></a></small>
      <div class="hidden name"><code>geuvadis.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Perform gene set analysis on the result of differential expression using linear (mixed) modeling with <a href="https://doi.org/10.1093/bioinformatics/btaa687">dream</a> by considering the correlation between gene expression traits. This package implements the <a href="https://www.rdocumentation.org/packages/limma/versions/3.28.14/topics/camera">camera</a> method from the <a href="https://bioconductor.org/packages/limma/">limma</a> package proposed by <a href="https://doi.org/10.1093/nar/gks461">Wu and Smyth (2012)</a>. <code>zenith()</code> is a simple extension of <code>camera()</code> to be compatible with linear (mixed) models implemented in <code>dream()</code>.</p>
    </div>
    
<div id="load-and-process-rna-seq-data" class="section level2">
<h2>Load and process RNA-seq data</h2>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(recount)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(variancePartition)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(edgeR)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(zenith)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(EnrichmentBrowser)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GSEABase)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="co"># For ENSEMBL id ENSG00000279457.4, return ENSG00000279457</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a>trim_ensembl_ids <span class="ot">=</span> <span class="cf">function</span>(x){</span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">gsub</span>(<span class="st">&quot;(.*)</span><span class="sc">\\</span><span class="st">.(.*)&quot;</span>, <span class="st">&quot;</span><span class="sc">\\</span><span class="st">1&quot;</span>, x) </span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<div id="load-rna-seq-data-and-metadata" class="section level4">
<h4>Load RNA-seq data and metadata</h4>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Download GEUVADIS data from recount2</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span>( <span class="sc">!</span> <span class="fu">file.exists</span>(<span class="st">&quot;ERP001942/rse_gene.Rdata&quot;</span>) ){</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">download_study</span>(<span class="st">&#39;ERP001942&#39;</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the data</span></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">load</span>(<span class="fu">file.path</span>(<span class="st">&#39;ERP001942&#39;</span>, <span class="st">&#39;rse_gene.Rdata&#39;</span>))</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="co"># scale counts</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>rse <span class="ot">&lt;-</span> <span class="fu">scale_counts</span>(rse_gene)</span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a><span class="co"># get count data</span></span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>geneCounts <span class="ot">=</span> <span class="fu">assay</span>(rse)</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a><span class="co"># read and format metadata</span></span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>info <span class="ot">=</span> <span class="fu">read.table</span>(<span class="st">&quot;https://www.ebi.ac.uk/arrayexpress/files/E-GEUV-1/E-GEUV-1.sdrf.txt&quot;</span>, <span class="at">header=</span><span class="cn">TRUE</span>, <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(info) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;Comment</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(info))</span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(info) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;Characteristics</span><span class="sc">\\</span><span class="st">.&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(info))</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(info) <span class="ot">=</span> <span class="fu">gsub</span>(<span class="st">&quot;</span><span class="sc">\\</span><span class="st">.$&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">colnames</span>(info))</span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a><span class="co"># extract unique samples and drop a problematic sample</span></span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>info <span class="ot">=</span> <span class="fu">unique</span>(info[,<span class="sc">!</span><span class="fu">colnames</span>(info) <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;Scan.Name&#39;</span>, <span class="st">&#39;SUBMITTED_FILE_NAME&#39;</span>, <span class="st">&#39;FASTQ_URI&#39;</span>)])</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>info <span class="ot">=</span> info[info<span class="sc">$</span>ENA_RUN <span class="sc">!=</span> <span class="st">&#39;ERR188328&#39;</span>,]</span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a><span class="co"># merge sample information in colData(rse) and info</span></span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">=</span> <span class="fu">merge</span>(<span class="fu">colData</span>(rse), info, <span class="at">by.x=</span><span class="st">&quot;run&quot;</span>, <span class="at">by.y=</span><span class="st">&#39;ENA_RUN&#39;</span>)</span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>metadata <span class="ot">=</span> <span class="fu">data.frame</span>(metadata)</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(metadata) <span class="ot">=</span> metadata<span class="sc">$</span>run</span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a><span class="co"># keep RNA-seq samples in metadata</span></span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>geneCounts <span class="ot">=</span> geneCounts[,<span class="fu">colnames</span>(geneCounts) <span class="sc">%in%</span> metadata<span class="sc">$</span>run]</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>geneCounts <span class="ot">=</span> geneCounts[,<span class="fu">match</span>(metadata<span class="sc">$</span>run, <span class="fu">colnames</span>(geneCounts))]</span></code></pre></div>
</div>
<div id="rna-seq-processing" class="section level4">
<h4>RNA-seq processing</h4>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep only expressed genes</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>isexpr <span class="ot">=</span> <span class="fu">rowSums</span>(<span class="fu">cpm</span>(geneCounts)<span class="sc">&gt;</span><span class="dv">1</span>) <span class="sc">&gt;=</span> <span class="fl">0.2</span><span class="sc">*</span><span class="fu">ncol</span>(geneCounts)</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="co"># apply TMM normalization</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">=</span> <span class="fu">DGEList</span>( geneCounts[isexpr,] )</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a>dge <span class="ot">=</span> <span class="fu">calcNormFactors</span>(dge)</span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Create log2 counts per million for plotting</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>geneExpr <span class="ot">=</span> <span class="fu">cpm</span>(dge, <span class="at">log=</span><span class="cn">TRUE</span>)</span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co"># extract gene annotations</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>geneInfo <span class="ot">=</span> <span class="fu">data.frame</span>(<span class="fu">rowData</span>(rse))</span></code></pre></div>
</div>
<div id="confirm-sex-based-on-expression" class="section level4">
<h4>Confirm sex based on expression</h4>
<p>Plot expression of XIST and UTY<br />
We see a clear separation of sex based on XIST from chrX and UTY from chrY.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># extract ENSEMBL identifies for XIST and UTY</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>ens_XIST <span class="ot">=</span> geneInfo<span class="sc">$</span>gene_id[<span class="fu">which</span>(geneInfo<span class="sc">$</span>symbol <span class="sc">==</span> <span class="st">&quot;XIST&quot;</span>)]</span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>ens_UTY <span class="ot">=</span> geneInfo<span class="sc">$</span>gene_id[<span class="fu">which</span>(geneInfo<span class="sc">$</span>symbol <span class="sc">==</span> <span class="st">&quot;UTY&quot;</span>)]</span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">data.frame</span>(metadata, <span class="at">XIST=</span> geneExpr[ens_XIST,],</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>        <span class="at">UTY =</span> geneExpr[ens_UTY,])</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(XIST, UTY, <span class="at">color=</span>sex)) <span class="sc">+</span> <span class="fu">geom_point</span>() <span class="sc">+</span> <span class="fu">theme_bw</span>() <span class="sc">+</span> <span class="fu">theme</span>(<span class="at">aspect.ratio=</span><span class="dv">1</span>) <span class="sc">+</span> <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span>, <span class="st">&quot;blue&quot;</span>))</span></code></pre></div>
<p><img src="geuvadis_files/figure-html/plot.sex-1.png" width="480" /></p>
</div>
<div id="differential-expression-with-dream" class="section level4">
<h4>Differential expression with <code>dream</code></h4>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>form <span class="ot">=</span> <span class="er">~</span> sex <span class="sc">+</span> ancestry.category <span class="sc">+</span> laboratory</span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>vobj <span class="ot">=</span> <span class="fu">voomWithDreamWeights</span>(dge, form, metadata, <span class="at">plot=</span><span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="geuvadis_files/figure-html/diffExpr-1.png" width="480" /></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">dream</span>(vobj, form, metadata, <span class="at">computeResiduals=</span><span class="cn">TRUE</span>)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">=</span> <span class="fu">eBayes</span>(fit)</span></code></pre></div>
</div>
<div id="load-genesets" class="section level4">
<h4>Load genesets</h4>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load Gene Ontology database </span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>go.gs <span class="ot">=</span> <span class="fu">get_GeneOntology</span>()</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a><span class="co"># convert from GeneSetCollection to list used by camera and zenith</span></span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>geneSets_GO <span class="ot">=</span> <span class="fu">recodeToList</span>( go.gs )</span></code></pre></div>
</div>
</div>
<div id="zenith-gene-set-analysis" class="section level2">
<h2>Zenith gene set analysis</h2>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Map from Ensembl genes in geneSets_GO to </span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="co"># from trimmed Ensembl names from RNA-seq data </span></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>index <span class="ot">=</span> <span class="fu">ids2indices</span>( geneSets_GO, <span class="fu">trim_ensembl_ids</span>(<span class="fu">rownames</span>(fit)))</span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>   </span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="co"># filter by size of gene set</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>n_genes_in <span class="ot">=</span> <span class="dv">10</span></span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>index <span class="ot">=</span> index[<span class="fu">sapply</span>(index, length) <span class="sc">&gt;=</span> n_genes_in]</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Run camera</span></span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>dsgn <span class="ot">=</span> <span class="fu">model.matrix</span>( form, metadata)</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>res_camera <span class="ot">=</span> <span class="fu">camera</span>( vobj, index, dsgn, <span class="at">inter.gene.cor=</span><span class="cn">NA</span>, <span class="at">contrast=</span><span class="st">&quot;sexmale&quot;</span>)</span>
<span id="cb8-12"><a href="#cb8-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-13"><a href="#cb8-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Run zenith.  </span></span>
<span id="cb8-14"><a href="#cb8-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Note that zenith allows analysis of any dream model and </span></span>
<span id="cb8-15"><a href="#cb8-15" aria-hidden="true" tabindex="-1"></a><span class="co"># any coefficient or contrast defined in the model. </span></span>
<span id="cb8-16"><a href="#cb8-16" aria-hidden="true" tabindex="-1"></a>res_zenith <span class="ot">=</span> <span class="fu">zenith</span>( fit, <span class="st">&quot;sexmale&quot;</span>, index)</span></code></pre></div>
<div id="compare-camera-and-zenith" class="section level3">
<h3>Compare camera and zenith</h3>
<div id="results-from-camera" class="section level4">
<h4>Results from camera</h4>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>res_camera[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,] <span class="sc">%&gt;%</span> kable <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>) </span></code></pre></div>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:right;">
Correlation
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
GO0060765: regulation of androgen receptor signaling pathway
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
-0.0126289
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:right;">
0.0045249
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0050681: androgen receptor binding
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
-0.0049552
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000263
</td>
<td style="text-align:right;">
0.0713334
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0030521: androgen receptor signaling pathway
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
0.0073531
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000350
</td>
<td style="text-align:right;">
0.0713334
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0045495: pole plasm
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.0066687
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000638
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0001941: postsynaptic membrane organization
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
-0.0006421
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000702
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0003785: actin monomer binding
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
-0.0068675
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000971
</td>
<td style="text-align:right;">
0.0988046
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0097110: scaffold protein binding
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
0.0032819
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0001190
</td>
<td style="text-align:right;">
0.1038613
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0035176: social behavior
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.0496503
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0016764
</td>
<td style="text-align:right;">
0.9997348
</td>
</tr>
</tbody>
</table>
</div>
<div id="results-from-zenith" class="section level4">
<h4>Results from zenith</h4>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>res_zenith[<span class="dv">1</span><span class="sc">:</span><span class="dv">8</span>,] <span class="sc">%&gt;%</span> kable <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)     </span></code></pre></div>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:right;">
Correlation
</th>
<th style="text-align:right;">
delta
</th>
<th style="text-align:right;">
se
</th>
<th style="text-align:right;">
p.less
</th>
<th style="text-align:right;">
p.greater
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
GO0060765: regulation of androgen receptor signaling pathway
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
-0.0138619
</td>
<td style="text-align:right;">
2.394862
</td>
<td style="text-align:right;">
0.4769345
</td>
<td style="text-align:right;">
0.9999996
</td>
<td style="text-align:right;">
0.0000004
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0045249
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0030521: androgen receptor signaling pathway
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
0.0058599
</td>
<td style="text-align:right;">
1.793398
</td>
<td style="text-align:right;">
0.4213249
</td>
<td style="text-align:right;">
0.9999874
</td>
<td style="text-align:right;">
0.0000126
</td>
<td style="text-align:right;">
0.0000253
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0535272
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0050681: androgen receptor binding
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
-0.0055522
</td>
<td style="text-align:right;">
1.777379
</td>
<td style="text-align:right;">
0.4184603
</td>
<td style="text-align:right;">
0.9999869
</td>
<td style="text-align:right;">
0.0000131
</td>
<td style="text-align:right;">
0.0000263
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0535272
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0045495: pole plasm
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.0036462
</td>
<td style="text-align:right;">
2.722434
</td>
<td style="text-align:right;">
0.6744655
</td>
<td style="text-align:right;">
0.9999681
</td>
<td style="text-align:right;">
0.0000319
</td>
<td style="text-align:right;">
0.0000638
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0001941: postsynaptic membrane organization
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
-0.0019566
</td>
<td style="text-align:right;">
1.964205
</td>
<td style="text-align:right;">
0.4894403
</td>
<td style="text-align:right;">
0.9999649
</td>
<td style="text-align:right;">
0.0000351
</td>
<td style="text-align:right;">
0.0000702
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0003785: actin monomer binding
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
-0.0087844
</td>
<td style="text-align:right;">
2.035066
</td>
<td style="text-align:right;">
0.5174099
</td>
<td style="text-align:right;">
0.9999515
</td>
<td style="text-align:right;">
0.0000485
</td>
<td style="text-align:right;">
0.0000971
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0911182
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0097110: scaffold protein binding
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
0.0027498
</td>
<td style="text-align:right;">
1.458051
</td>
<td style="text-align:right;">
0.3724309
</td>
<td style="text-align:right;">
0.9999478
</td>
<td style="text-align:right;">
0.0000522
</td>
<td style="text-align:right;">
0.0001044
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0911182
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0035176: social behavior
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.0479135
</td>
<td style="text-align:right;">
1.996151
</td>
<td style="text-align:right;">
0.6253904
</td>
<td style="text-align:right;">
0.9992437
</td>
<td style="text-align:right;">
0.0007563
</td>
<td style="text-align:right;">
0.0015127
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.9997350
</td>
</tr>
</tbody>
</table>
<p>While there is a very slight difference in the results from the methods, this is attributable to the very slight difference in the correlation value computed. This appears to be due to an numerical issue with <code>camera</code>, but values are correct to two digits.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">=</span> <span class="fu">merge</span>(res_camera, res_zenith, <span class="at">by=</span><span class="st">&quot;row.names&quot;</span>)</span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(df<span class="sc">$</span>Correlation.x, df<span class="sc">$</span>Correlation.y, <span class="at">xlab=</span><span class="st">&quot;Correlation from zenith&quot;</span>, <span class="at">ylab=</span><span class="st">&quot;Correlation from camera&quot;</span>)</span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="dv">0</span>, <span class="dv">1</span>, <span class="at">col=</span><span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="geuvadis_files/figure-html/compare-1.png" width="480" /></p>
</div>
</div>
</div>
<div id="analysis-with-msigdb" class="section level2">
<h2>Analysis with MSigDB</h2>
<div id="load-msigdb-and-perform-zenith-analysis" class="section level4">
<h4>Load <a href="https://www.gsea-msigdb.org/gsea/msigdb/index.jsp">MSigDB</a> and perform zenith analysis</h4>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>gs_msigdb <span class="ot">=</span> <span class="fu">get_MSigDB</span>()</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Map from Ensembl genes in gs_msigdb to </span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="co"># from trimmed Ensembl names from RNA-seq data </span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>index <span class="ot">=</span> <span class="fu">ids2indices</span>( <span class="fu">recodeToList</span>(gs_msigdb), <span class="fu">trim_ensembl_ids</span>(<span class="fu">rownames</span>(fit)))</span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a><span class="co"># filter by size of gene set</span></span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>n_genes_in <span class="ot">=</span> <span class="dv">10</span></span>
<span id="cb12-9"><a href="#cb12-9" aria-hidden="true" tabindex="-1"></a>index <span class="ot">=</span> index[<span class="fu">sapply</span>(index, length) <span class="sc">&gt;=</span> n_genes_in]</span>
<span id="cb12-10"><a href="#cb12-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb12-11"><a href="#cb12-11" aria-hidden="true" tabindex="-1"></a> <span class="co"># zenith analysis of MSigDB using square test statistics</span></span>
<span id="cb12-12"><a href="#cb12-12" aria-hidden="true" tabindex="-1"></a>res_zenith_msigdb <span class="ot">=</span> <span class="fu">zenith</span>( fit, <span class="st">&quot;sexmale&quot;</span>, index)</span></code></pre></div>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>res_zenith_msigdb[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>,] <span class="sc">%&gt;%</span> kable <span class="sc">%&gt;%</span> <span class="fu">kable_styling</span>(<span class="at">full_width =</span> <span class="cn">FALSE</span>)  </span></code></pre></div>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:right;">
Correlation
</th>
<th style="text-align:right;">
delta
</th>
<th style="text-align:right;">
se
</th>
<th style="text-align:right;">
p.less
</th>
<th style="text-align:right;">
p.greater
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
FDR
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
M5126_PYEON_CANCER_HEAD_AND_NECK_VS_CERVICAL_DN
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
0.1418496
</td>
<td style="text-align:right;">
19.224970
</td>
<td style="text-align:right;">
0.9422304
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
M16091_DISTECHE_ESCAPED_FROM_X_INACTIVATION
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
0.0715801
</td>
<td style="text-align:right;">
-14.671145
</td>
<td style="text-align:right;">
0.7922187
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
M39089_ZHONG_PFC_C2_SOX5_BCL11B_POS_EXCITATORY_NEURON
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
0.0046050
</td>
<td style="text-align:right;">
6.648676
</td>
<td style="text-align:right;">
0.5491522
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
M7727_RICKMAN_HEAD_AND_NECK_CANCER_B
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
0.0053964
</td>
<td style="text-align:right;">
4.895289
</td>
<td style="text-align:right;">
0.5113755
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
M34944_HP_OLIGOSPERMIA
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.0177363
</td>
<td style="text-align:right;">
5.544998
</td>
<td style="text-align:right;">
0.6309138
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000000
</td>
</tr>
<tr>
<td style="text-align:left;">
M32591_MIR151A_5P_MIR151B
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.0163958
</td>
<td style="text-align:right;">
4.783951
</td>
<td style="text-align:right;">
0.7217704
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000003
</td>
</tr>
<tr>
<td style="text-align:left;">
M22112_chrXp22
</td>
<td style="text-align:right;">
74
</td>
<td style="text-align:right;">
0.0288466
</td>
<td style="text-align:right;">
-2.886132
</td>
<td style="text-align:right;">
0.4356068
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.0000003
</td>
</tr>
<tr>
<td style="text-align:left;">
M22115_chrXq13
</td>
<td style="text-align:right;">
45
</td>
<td style="text-align:right;">
0.0272969
</td>
<td style="text-align:right;">
-3.048338
</td>
<td style="text-align:right;">
0.4708069
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.0000008
</td>
</tr>
<tr>
<td style="text-align:left;">
M37972_HP_NON_OBSTRUCTIVE_AZOOSPERMIA
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
0.0424284
</td>
<td style="text-align:right;">
4.306009
</td>
<td style="text-align:right;">
0.6692211
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000009
</td>
</tr>
<tr>
<td style="text-align:left;">
M6596_GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_DN
</td>
<td style="text-align:right;">
119
</td>
<td style="text-align:right;">
0.0228115
</td>
<td style="text-align:right;">
2.193358
</td>
<td style="text-align:right;">
0.3747157
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
0.0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000226
</td>
</tr>
<tr>
<td style="text-align:left;">
M11031_CHEOK_RESPONSE_TO_MERCAPTOPURINE_AND_LD_MTX_UP
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.1047066
</td>
<td style="text-align:right;">
5.084481
</td>
<td style="text-align:right;">
0.9387448
</td>
<td style="text-align:right;">
1.0000000
</td>
<td style="text-align:right;">
0.0000000
</td>
<td style="text-align:right;">
1.0e-07
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0002196
</td>
</tr>
<tr>
<td style="text-align:left;">
M8555_MODULE_489
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
0.0257540
</td>
<td style="text-align:right;">
3.819485
</td>
<td style="text-align:right;">
0.7207211
</td>
<td style="text-align:right;">
0.9999999
</td>
<td style="text-align:right;">
0.0000001
</td>
<td style="text-align:right;">
2.0e-07
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0003694
</td>
</tr>
<tr>
<td style="text-align:left;">
M18555_MODULE_330
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
0.0058980
</td>
<td style="text-align:right;">
3.393817
</td>
<td style="text-align:right;">
0.6615542
</td>
<td style="text-align:right;">
0.9999998
</td>
<td style="text-align:right;">
0.0000002
</td>
<td style="text-align:right;">
4.0e-07
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0008073
</td>
</tr>
<tr>
<td style="text-align:left;">
M12117_GOBP_REGULATION_OF_ANDROGEN_RECEPTOR_SIGNALING_PATHWAY
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
-0.0138619
</td>
<td style="text-align:right;">
2.394862
</td>
<td style="text-align:right;">
0.4769345
</td>
<td style="text-align:right;">
0.9999996
</td>
<td style="text-align:right;">
0.0000004
</td>
<td style="text-align:right;">
7.0e-07
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0012681
</td>
</tr>
<tr>
<td style="text-align:left;">
M1659_CARDOSO_RESPONSE_TO_GAMMA_RADIATION_AND_3AB
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
-0.0267058
</td>
<td style="text-align:right;">
2.963385
</td>
<td style="text-align:right;">
0.5914451
</td>
<td style="text-align:right;">
0.9999996
</td>
<td style="text-align:right;">
0.0000004
</td>
<td style="text-align:right;">
8.0e-07
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0012681
</td>
</tr>
<tr>
<td style="text-align:left;">
M5384_GSE3982_MEMORY_CD4_TCELL_VS_BCELL_UP
</td>
<td style="text-align:right;">
143
</td>
<td style="text-align:right;">
0.0142444
</td>
<td style="text-align:right;">
1.522163
</td>
<td style="text-align:right;">
0.3099245
</td>
<td style="text-align:right;">
0.9999994
</td>
<td style="text-align:right;">
0.0000006
</td>
<td style="text-align:right;">
1.3e-06
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0017352
</td>
</tr>
<tr>
<td style="text-align:left;">
M22110_chrXp11
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
0.0173072
</td>
<td style="text-align:right;">
-1.709990
</td>
<td style="text-align:right;">
0.3486447
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:right;">
0.9999993
</td>
<td style="text-align:right;">
1.3e-06
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.0017352
</td>
</tr>
<tr>
<td style="text-align:left;">
M15380_KORKOLA_EMBRYONIC_CARCINOMA_VS_SEMINOMA_DN
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
-0.0070549
</td>
<td style="text-align:right;">
-3.146648
</td>
<td style="text-align:right;">
0.6429526
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:right;">
0.9999993
</td>
<td style="text-align:right;">
1.4e-06
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.0017352
</td>
</tr>
<tr>
<td style="text-align:left;">
M14833_GAUSSMANN_MLL_AF4_FUSION_TARGETS_B_UP
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
0.0065918
</td>
<td style="text-align:right;">
-2.734085
</td>
<td style="text-align:right;">
0.5588630
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:right;">
0.9999993
</td>
<td style="text-align:right;">
1.4e-06
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.0017352
</td>
</tr>
<tr>
<td style="text-align:left;">
M32473_MIR3135A
</td>
<td style="text-align:right;">
29
</td>
<td style="text-align:right;">
0.0160716
</td>
<td style="text-align:right;">
2.330129
</td>
<td style="text-align:right;">
0.4768045
</td>
<td style="text-align:right;">
0.9999993
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:right;">
1.4e-06
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0017352
</td>
</tr>
</tbody>
</table>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc">
      <h2 data-toc-skip>Contents</h2>
    </nav>
      </div>

</div>



      <footer>
      <div class="copyright">
  <p>Developed by Gabriel Hoffman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>
