<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="zenith">
<title>Zenith gene set testing after dream analysis • zenith</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Zenith gene set testing after dream analysis">
<meta property="og:description" content="zenith">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">zenith</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.8</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/zenith.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/DiseaseNeuroGenomics/zenith/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">



<script src="zenith_files/kePrint-0.0.1/kePrint.js"></script><link href="zenith_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <main id="main"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Zenith gene set testing after dream analysis</h1>
                        <h4 data-toc-skip class="author"><a href="http://gabrielhoffman.github.io" class="external-link">Gabriel Hoffman</a></h4>
            <address class="author_afil">
      <p>Icahn School of Medicine at Mount Sinai, New York</p>
<br><small class="dont-index">Source: <a href="https://github.com/DiseaseNeuroGenomics/zenith/blob/HEAD/vignettes/zenith.Rmd" class="external-link"><code>vignettes/zenith.Rmd</code></a></small>
      <div class="d-none name"><code>zenith.Rmd</code></div>
    </address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Perform gene set analysis on the result of differential expression using linear (mixed) modeling with <a href="https://doi.org/10.1093/bioinformatics/btaa687" class="external-link">dream</a> by considering the correlation between gene expression traits. This package implements the <a href="https://www.rdocumentation.org/packages/limma/versions/3.28.14/topics/camera" class="external-link">camera</a> method from the <a href="https://bioconductor.org/packages/limma/" class="external-link">limma</a> package proposed by <a href="https://doi.org/10.1093/nar/gks461" class="external-link">Wu and Smyth (2012)</a>. <code><a href="../reference/zenith.html">zenith()</a></code> is a simple extension of <code><a href="https://rdrr.io/pkg/limma/man/camera.html" class="external-link">camera()</a></code> to be compatible with linear (mixed) models implemented in <code><a href="http://gabrielhoffman.github.io/variancePartition/reference/dream-method.html" class="external-link">dream()</a></code>.</p>
    </div>
    
<div class="section level2">
<h2 id="loading-gene-sets">Loading gene sets<a class="anchor" aria-label="anchor" href="#loading-gene-sets"></a>
</h2>
<div class="section level3">
<h3 id="supported-gene-set-databases">Supported gene set databases<a class="anchor" aria-label="anchor" href="#supported-gene-set-databases"></a>
</h3>
<p><a href="https://bioconductor.org/packages/EnrichmentBrowser/" class="external-link">EnrichmentBrowser</a> provides access to a range of gene set databases. Here are some shortcuts to load common databases:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://DiseaseNeuroGenomics.github.io/zenith" class="external-link">zenith</a></span><span class="op">)</span>

<span class="co"># MSigDB</span>
<span class="va">gs.msigdb</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_MSigDB.html">get_MSigDB</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Gene Ontology</span>
<span class="va">gs.go</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_GeneOntology.html">get_GeneOntology</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>For more detailed control including alternate gene identifiers (i.e. ENSEMBL, ENTREZ) or species (i.e. hsa, mmu)</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">EnrichmentBrowser</span><span class="op">)</span>

<span class="co"># KEGG</span>
<span class="va">gs.kegg</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/getGenesets.html" class="external-link">getGenesets</a></span><span class="op">(</span>  org <span class="op">=</span> <span class="st">"hsa"</span>, 
              db <span class="op">=</span> <span class="st">"kegg"</span>, 
              gene.id.type <span class="op">=</span> <span class="st">"ENSEMBL"</span>, 
              return.type <span class="op">=</span> <span class="st">"GeneSetCollection"</span><span class="op">)</span>

<span class="co">## ENRICHR resource</span>
<span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/getGenesets.html" class="external-link">showAvailableCollections</a></span><span class="op">(</span> org <span class="op">=</span> <span class="st">"hsa"</span>, db <span class="op">=</span> <span class="st">"enrichr"</span><span class="op">)</span>

<span class="co"># Allen_Brain_Atlas_10x_scRNA_2021</span>
<span class="va">gs.allen</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/getGenesets.html" class="external-link">getGenesets</a></span><span class="op">(</span>  org <span class="op">=</span> <span class="st">"hsa"</span>, 
              db <span class="op">=</span> <span class="st">"enrichr"</span>, 
              lib <span class="op">=</span> <span class="st">"Allen_Brain_Atlas_10x_scRNA_2021"</span>,
              gene.id.type <span class="op">=</span> <span class="st">"ENSEMBL"</span>, 
              return.type <span class="op">=</span> <span class="st">"GeneSetCollection"</span><span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="integration-with-dream">Integration with dream<a class="anchor" aria-label="anchor" href="#integration-with-dream"></a>
</h2>
<div class="section level3">
<h3 id="load-and-process-rna-seq-data">Load and process RNA-seq data<a class="anchor" aria-label="anchor" href="#load-and-process-rna-seq-data"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/leekgroup/recount" class="external-link">recount</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/variancePartition" class="external-link">variancePartition</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR" class="external-link">edgeR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">limma</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://DiseaseNeuroGenomics.github.io/zenith" class="external-link">zenith</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">EnrichmentBrowser</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">GSEABase</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://haozhu233.github.io/kableExtra/" class="external-link">kableExtra</a></span><span class="op">)</span>

<span class="co"># For ENSEMBL id ENSG00000279457.4, return ENSG00000279457</span>
<span class="va">trim_ensembl_ids</span> <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(.*)\\.(.*)"</span>, <span class="st">"\\1"</span>, <span class="va">x</span><span class="op">)</span> 
<span class="op">}</span></code></pre></div>
<div class="section level5">
<h5 id="load-rna-seq-data-and-metadata">Load RNA-seq data and metadata<a class="anchor" aria-label="anchor" href="#load-rna-seq-data-and-metadata"></a>
</h5>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Download GEUVADIS data from recount2</span>
<span class="kw">if</span><span class="op">(</span> <span class="op">!</span> <span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.exists</a></span><span class="op">(</span><span class="st">"ERP001942/rse_gene.Rdata"</span><span class="op">)</span> <span class="op">)</span><span class="op">{</span>
  <span class="fu"><a href="https://rdrr.io/pkg/recount/man/download_study.html" class="external-link">download_study</a></span><span class="op">(</span><span class="st">'ERP001942'</span><span class="op">)</span>
<span class="op">}</span>

<span class="co"># Load the data</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">'ERP001942'</span>, <span class="st">'rse_gene.Rdata'</span><span class="op">)</span><span class="op">)</span>

<span class="co"># scale counts</span>
<span class="va">rse</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/recount/man/scale_counts.html" class="external-link">scale_counts</a></span><span class="op">(</span><span class="va">rse_gene</span><span class="op">)</span>

<span class="co"># get count data</span>
<span class="va">geneCounts</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">rse</span><span class="op">)</span>

<span class="co"># read and format metadata</span>
<span class="va">info</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="st">"https://www.ebi.ac.uk/arrayexpress/files/E-GEUV-1/E-GEUV-1.sdrf.txt"</span>, header<span class="op">=</span><span class="cn">TRUE</span>, sep<span class="op">=</span><span class="st">"\t"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"Comment\\."</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"Characteristics\\."</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"\\.$"</span>, <span class="st">""</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span><span class="op">)</span>

<span class="co"># extract unique samples and drop a problematic sample</span>
<span class="va">info</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">info</span><span class="op">[</span>,<span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">info</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Scan.Name'</span>, <span class="st">'SUBMITTED_FILE_NAME'</span>, <span class="st">'FASTQ_URI'</span><span class="op">)</span><span class="op">]</span><span class="op">)</span>
<span class="va">info</span> <span class="op">=</span> <span class="va">info</span><span class="op">[</span><span class="va">info</span><span class="op">$</span><span class="va">ENA_RUN</span> <span class="op">!=</span> <span class="st">'ERR188328'</span>,<span class="op">]</span>

<span class="co"># merge sample information in colData(rse) and info</span>
<span class="va">metadata</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">rse</span><span class="op">)</span>, <span class="va">info</span>, by.x<span class="op">=</span><span class="st">"run"</span>, by.y<span class="op">=</span><span class="st">'ENA_RUN'</span><span class="op">)</span>
<span class="va">metadata</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span> <span class="op">=</span> <span class="va">metadata</span><span class="op">$</span><span class="va">run</span>

<span class="co"># keep RNA-seq samples in metadata</span>
<span class="va">geneCounts</span> <span class="op">=</span> <span class="va">geneCounts</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">metadata</span><span class="op">$</span><span class="va">run</span><span class="op">]</span>
<span class="va">geneCounts</span> <span class="op">=</span> <span class="va">geneCounts</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">$</span><span class="va">run</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></code></pre></div>
</div>
<div class="section level5">
<h5 id="rna-seq-processing">RNA-seq processing<a class="anchor" aria-label="anchor" href="#rna-seq-processing"></a>
</h5>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># keep only expressed genes</span>
<span class="va">isexpr</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/colSums.html" class="external-link">rowSums</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span><span class="op">&gt;</span><span class="fl">1</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="fl">0.2</span><span class="op">*</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">geneCounts</span><span class="op">)</span>

<span class="co"># apply TMM normalization</span>
<span class="va">dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span> <span class="va">geneCounts</span><span class="op">[</span><span class="va">isexpr</span>,<span class="op">]</span> <span class="op">)</span>
<span class="va">dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">dge</span><span class="op">)</span>

<span class="co"># Create log2 counts per million for plotting</span>
<span class="va">geneExpr</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">dge</span>, log<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>

<span class="co"># extract gene annotations</span>
<span class="va">geneInfo</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">rse</span><span class="op">)</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level5">
<h5 id="confirm-sex-based-on-expression">Confirm sex based on expression<a class="anchor" aria-label="anchor" href="#confirm-sex-based-on-expression"></a>
</h5>
<p>Plot expression of XIST and UTY<br>
We see a clear separation of sex based on XIST from chrX and UTY from chrY.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># extract ENSEMBL identifies for XIST and UTY</span>
<span class="va">ens_XIST</span> <span class="op">=</span> <span class="va">geneInfo</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">geneInfo</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"XIST"</span><span class="op">)</span><span class="op">]</span>
<span class="va">ens_UTY</span> <span class="op">=</span> <span class="va">geneInfo</span><span class="op">$</span><span class="va">gene_id</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/which.html" class="external-link">which</a></span><span class="op">(</span><span class="va">geneInfo</span><span class="op">$</span><span class="va">symbol</span> <span class="op">==</span> <span class="st">"UTY"</span><span class="op">)</span><span class="op">]</span>

<span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span><span class="va">metadata</span>, XIST<span class="op">=</span> <span class="va">geneExpr</span><span class="op">[</span><span class="va">ens_XIST</span>,<span class="op">]</span>,
        UTY <span class="op">=</span> <span class="va">geneExpr</span><span class="op">[</span><span class="va">ens_UTY</span>,<span class="op">]</span><span class="op">)</span>

<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">XIST</span>, <span class="va">UTY</span>, color<span class="op">=</span><span class="va">sex</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>aspect.ratio<span class="op">=</span><span class="fl">1</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"red"</span>, <span class="st">"blue"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="zenith_files/figure-html/plot.sex-1.png" width="480"></p>
</div>
<div class="section level5">
<h5 id="differential-expression-with-dream">Differential expression with <code>dream</code><a class="anchor" aria-label="anchor" href="#differential-expression-with-dream"></a>
</h5>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">form</span> <span class="op">=</span> <span class="op">~</span> <span class="va">sex</span> <span class="op">+</span> <span class="va">ancestry.category</span> <span class="op">+</span> <span class="va">laboratory</span>
<span class="va">vobj</span> <span class="op">=</span> <span class="fu"><a href="http://gabrielhoffman.github.io/variancePartition/reference/voomWithDreamWeights.html" class="external-link">voomWithDreamWeights</a></span><span class="op">(</span><span class="va">dge</span>, <span class="va">form</span>, <span class="va">metadata</span>, plot<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p><img src="zenith_files/figure-html/diffExpr-1.png" width="480"></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="http://gabrielhoffman.github.io/variancePartition/reference/dream-method.html" class="external-link">dream</a></span><span class="op">(</span><span class="va">vobj</span>, <span class="va">form</span>, <span class="va">metadata</span>, computeResiduals<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level5">
<h5 id="load-genesets">Load genesets<a class="anchor" aria-label="anchor" href="#load-genesets"></a>
</h5>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Load Gene Ontology database </span>
<span class="va">go.gs</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_GeneOntology.html">get_GeneOntology</a></span><span class="op">(</span><span class="op">)</span>
   
<span class="co"># convert from GeneSetCollection to list used by camera and zenith</span>
<span class="va">geneSets_GO</span> <span class="op">=</span> <span class="fu"><a href="../reference/recodeToList.html">recodeToList</a></span><span class="op">(</span> <span class="va">go.gs</span> <span class="op">)</span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="zenith-gene-set-analysis">Zenith gene set analysis<a class="anchor" aria-label="anchor" href="#zenith-gene-set-analysis"></a>
</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Map from Ensembl genes in geneSets_GO to </span>
<span class="co"># from trimmed Ensembl names from RNA-seq data </span>
<span class="va">index</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ids2indices.html" class="external-link">ids2indices</a></span><span class="op">(</span> <span class="va">geneSets_GO</span>, <span class="fu">trim_ensembl_ids</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>
   
<span class="co"># filter by size of gene set</span>
<span class="va">n_genes_in</span> <span class="op">=</span> <span class="fl">10</span>
<span class="va">index</span> <span class="op">=</span> <span class="va">index</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">index</span>, <span class="va">length</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">n_genes_in</span><span class="op">]</span>

<span class="co"># Run camera</span>
<span class="va">dsgn</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span> <span class="va">form</span>, <span class="va">metadata</span><span class="op">)</span>
<span class="va">res_camera</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/camera.html" class="external-link">camera</a></span><span class="op">(</span> <span class="va">vobj</span>, <span class="va">index</span>, <span class="va">dsgn</span>, inter.gene.cor<span class="op">=</span><span class="cn">NA</span>, contrast<span class="op">=</span><span class="st">"sexmale"</span><span class="op">)</span>

<span class="co"># Run zenith.  </span>
<span class="co"># Note that zenith allows analysis of any dream model and </span>
<span class="co"># any coefficient or contrast defined in the model. </span>
<span class="co"># Setting inter.gene.cor=NA estimates the correlation from the model</span>
<span class="va">res_zenith</span> <span class="op">=</span> <span class="fu"><a href="../reference/zenith.html">zenith</a></span><span class="op">(</span> <span class="va">fit</span>, <span class="st">"sexmale"</span>, <span class="va">index</span>, inter.gene.cor<span class="op">=</span><span class="cn">NA</span><span class="op">)</span></code></pre></div>
<div class="section level4">
<h4 id="compare-camera-and-zenith">Compare camera and zenith<a class="anchor" aria-label="anchor" href="#compare-camera-and-zenith"></a>
</h4>
<div class="section level5">
<h5 id="results-from-camera">Results from camera<a class="anchor" aria-label="anchor" href="#results-from-camera"></a>
</h5>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_camera</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">kable</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html" class="external-link">kable_styling</a></span><span class="op">(</span>full_width <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> </code></pre></div>
<table class="table table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:right;">
Correlation
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:right;">
FDR
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GO0060765: regulation of androgen receptor signaling pathway
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
-0.0126289
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:right;">
0.0045249
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0050681: androgen receptor binding
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
-0.0049552
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000263
</td>
<td style="text-align:right;">
0.0713334
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0030521: androgen receptor signaling pathway
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
0.0073531
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000350
</td>
<td style="text-align:right;">
0.0713334
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0045495: pole plasm
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.0066687
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000638
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0001941: postsynaptic membrane organization
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
-0.0006421
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000702
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0003785: actin monomer binding
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
-0.0068675
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0000971
</td>
<td style="text-align:right;">
0.0988046
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0097110: scaffold protein binding
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
0.0032819
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0001190
</td>
<td style="text-align:right;">
0.1038613
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0035176: social behavior
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.0496503
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0016764
</td>
<td style="text-align:right;">
0.9997348
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level5">
<h5 id="results-from-zenith">Results from zenith<a class="anchor" aria-label="anchor" href="#results-from-zenith"></a>
</h5>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_zenith</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">8</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">kable</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html" class="external-link">kable_styling</a></span><span class="op">(</span>full_width <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>     </code></pre></div>
<table class="table table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:right;">
Correlation
</th>
<th style="text-align:right;">
delta
</th>
<th style="text-align:right;">
se
</th>
<th style="text-align:right;">
p.less
</th>
<th style="text-align:right;">
p.greater
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
FDR
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
GO0060765: regulation of androgen receptor signaling pathway
</td>
<td style="text-align:right;">
20
</td>
<td style="text-align:right;">
-0.0138619
</td>
<td style="text-align:right;">
2.394862
</td>
<td style="text-align:right;">
0.4769345
</td>
<td style="text-align:right;">
0.9999996
</td>
<td style="text-align:right;">
0.0000004
</td>
<td style="text-align:right;">
0.0000007
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0045249
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0030521: androgen receptor signaling pathway
</td>
<td style="text-align:right;">
30
</td>
<td style="text-align:right;">
0.0058599
</td>
<td style="text-align:right;">
1.793398
</td>
<td style="text-align:right;">
0.4213249
</td>
<td style="text-align:right;">
0.9999874
</td>
<td style="text-align:right;">
0.0000126
</td>
<td style="text-align:right;">
0.0000253
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0535272
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0050681: androgen receptor binding
</td>
<td style="text-align:right;">
26
</td>
<td style="text-align:right;">
-0.0055522
</td>
<td style="text-align:right;">
1.777379
</td>
<td style="text-align:right;">
0.4184603
</td>
<td style="text-align:right;">
0.9999869
</td>
<td style="text-align:right;">
0.0000131
</td>
<td style="text-align:right;">
0.0000263
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0535272
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0045495: pole plasm
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
-0.0036462
</td>
<td style="text-align:right;">
2.722434
</td>
<td style="text-align:right;">
0.6744655
</td>
<td style="text-align:right;">
0.9999681
</td>
<td style="text-align:right;">
0.0000319
</td>
<td style="text-align:right;">
0.0000638
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0001941: postsynaptic membrane organization
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
-0.0019566
</td>
<td style="text-align:right;">
1.964205
</td>
<td style="text-align:right;">
0.4894403
</td>
<td style="text-align:right;">
0.9999649
</td>
<td style="text-align:right;">
0.0000351
</td>
<td style="text-align:right;">
0.0000702
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0857104
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0003785: actin monomer binding
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
-0.0087844
</td>
<td style="text-align:right;">
2.035066
</td>
<td style="text-align:right;">
0.5174099
</td>
<td style="text-align:right;">
0.9999515
</td>
<td style="text-align:right;">
0.0000485
</td>
<td style="text-align:right;">
0.0000971
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0911182
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0097110: scaffold protein binding
</td>
<td style="text-align:right;">
36
</td>
<td style="text-align:right;">
0.0027498
</td>
<td style="text-align:right;">
1.458051
</td>
<td style="text-align:right;">
0.3724309
</td>
<td style="text-align:right;">
0.9999478
</td>
<td style="text-align:right;">
0.0000522
</td>
<td style="text-align:right;">
0.0001044
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.0911182
</td>
</tr>
<tr>
<td style="text-align:left;">
GO0035176: social behavior
</td>
<td style="text-align:right;">
25
</td>
<td style="text-align:right;">
0.0479135
</td>
<td style="text-align:right;">
1.996151
</td>
<td style="text-align:right;">
0.6253904
</td>
<td style="text-align:right;">
0.9992437
</td>
<td style="text-align:right;">
0.0007563
</td>
<td style="text-align:right;">
0.0015127
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.9997350
</td>
</tr>
</tbody>
</table>
<p>While there is a very slight difference in the results from the methods, this is attributable to the very slight difference in the correlation value computed. This appears to be due to an numerical issue with <code>camera</code>, but values are correct to two digits.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">res_camera</span>, <span class="va">res_zenith</span>, by<span class="op">=</span><span class="st">"row.names"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">df</span><span class="op">$</span><span class="va">Correlation.x</span>, <span class="va">df</span><span class="op">$</span><span class="va">Correlation.y</span>, xlab<span class="op">=</span><span class="st">"Correlation from zenith"</span>, ylab<span class="op">=</span><span class="st">"Correlation from camera"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, col<span class="op">=</span><span class="st">"red"</span><span class="op">)</span></code></pre></div>
<p><img src="zenith_files/figure-html/compare-1.png" width="480"></p>
</div>
</div>
</div>
<div class="section level3">
<h3 id="analysis-with-msigdb">Analysis with MSigDB<a class="anchor" aria-label="anchor" href="#analysis-with-msigdb"></a>
</h3>
<div class="section level5">
<h5 id="load-msigdb-and-perform-zenith-analysis">Load <a href="https://www.gsea-msigdb.org/gsea/msigdb/index.jsp" class="external-link">MSigDB</a> and perform zenith analysis<a class="anchor" aria-label="anchor" href="#load-msigdb-and-perform-zenith-analysis"></a>
</h5>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gs_msigdb</span> <span class="op">=</span> <span class="fu"><a href="../reference/get_MSigDB.html">get_MSigDB</a></span><span class="op">(</span><span class="op">)</span>

<span class="co"># Map from Ensembl genes in gs_msigdb to </span>
<span class="co"># from trimmed Ensembl names from RNA-seq data </span>
<span class="va">index</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ids2indices.html" class="external-link">ids2indices</a></span><span class="op">(</span> <span class="fu"><a href="../reference/recodeToList.html">recodeToList</a></span><span class="op">(</span><span class="va">gs_msigdb</span><span class="op">)</span>, <span class="fu">trim_ensembl_ids</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co"># filter by size of gene set</span>
<span class="va">n_genes_in</span> <span class="op">=</span> <span class="fl">10</span>
<span class="va">index</span> <span class="op">=</span> <span class="va">index</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">index</span>, <span class="va">length</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">n_genes_in</span><span class="op">]</span>

 <span class="co"># zenith analysis of MSigDB using square test statistics</span>
<span class="va">res_zenith_msigdb</span> <span class="op">=</span> <span class="fu"><a href="../reference/zenith.html">zenith</a></span><span class="op">(</span> <span class="va">fit</span>, <span class="st">"sexmale"</span>, <span class="va">index</span><span class="op">)</span></code></pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_zenith_msigdb</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">20</span>,<span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="va">kable</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/kableExtra/man/kable_styling.html" class="external-link">kable_styling</a></span><span class="op">(</span>full_width <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>  </code></pre></div>
<table class="table table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
NGenes
</th>
<th style="text-align:right;">
Correlation
</th>
<th style="text-align:right;">
delta
</th>
<th style="text-align:right;">
se
</th>
<th style="text-align:right;">
p.less
</th>
<th style="text-align:right;">
p.greater
</th>
<th style="text-align:right;">
PValue
</th>
<th style="text-align:left;">
Direction
</th>
<th style="text-align:right;">
FDR
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
M5126_PYEON_CANCER_HEAD_AND_NECK_VS_CERVICAL_DN
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
19.224970
</td>
<td style="text-align:right;">
0.6067280
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M16091_DISTECHE_ESCAPED_FROM_X_INACTIVATION
</td>
<td style="text-align:right;">
13
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-14.671145
</td>
<td style="text-align:right;">
0.6150034
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M39089_ZHONG_PFC_C2_SOX5_BCL11B_POS_EXCITATORY_NEURON
</td>
<td style="text-align:right;">
16
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
6.648676
</td>
<td style="text-align:right;">
0.5695400
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M7727_RICKMAN_HEAD_AND_NECK_CANCER_B
</td>
<td style="text-align:right;">
19
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
4.895289
</td>
<td style="text-align:right;">
0.5303168
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M34944_HP_OLIGOSPERMIA
</td>
<td style="text-align:right;">
14
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
5.544998
</td>
<td style="text-align:right;">
0.6046005
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M22112_chrXp22
</td>
<td style="text-align:right;">
74
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-2.886132
</td>
<td style="text-align:right;">
0.3252910
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M22115_chrXq13
</td>
<td style="text-align:right;">
45
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-3.048338
</td>
<td style="text-align:right;">
0.3809299
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M37972_HP_NON_OBSTRUCTIVE_AZOOSPERMIA
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
4.306009
</td>
<td style="text-align:right;">
0.5563521
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M6596_GSE5099_CLASSICAL_M1_VS_ALTERNATIVE_M2_MACROPHAGE_DN
</td>
<td style="text-align:right;">
119
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
2.193358
</td>
<td style="text-align:right;">
0.2881369
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M11031_CHEOK_RESPONSE_TO_MERCAPTOPURINE_AND_LD_MTX_UP
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
5.084481
</td>
<td style="text-align:right;">
0.7033105
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
0.00e+00
</td>
</tr>
<tr>
<td style="text-align:left;">
M32591_MIR151A_5P_MIR151B
</td>
<td style="text-align:right;">
10
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
4.783951
</td>
<td style="text-align:right;">
0.7034449
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
1.00e-07
</td>
</tr>
<tr>
<td style="text-align:left;">
M2778_TAVAZOIE_METASTASIS
</td>
<td style="text-align:right;">
54
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
2.370876
</td>
<td style="text-align:right;">
0.3587496
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
2.00e-07
</td>
</tr>
<tr>
<td style="text-align:left;">
M37270_HP_ABNORMAL_SPERMATOGENESIS
</td>
<td style="text-align:right;">
72
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
2.072099
</td>
<td style="text-align:right;">
0.3285087
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
1.30e-06
</td>
</tr>
<tr>
<td style="text-align:left;">
M6669_HEDENFALK_BREAST_CANCER_BRACX_UP
</td>
<td style="text-align:right;">
17
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-3.256621
</td>
<td style="text-align:right;">
0.5568357
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
1.66e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M22110_chrXp11
</td>
<td style="text-align:right;">
104
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-1.709990
</td>
<td style="text-align:right;">
0.2979060
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
2.81e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M17751_chr19p13
</td>
<td style="text-align:right;">
425
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-1.353669
</td>
<td style="text-align:right;">
0.2362841
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
2.81e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M8555_MODULE_489
</td>
<td style="text-align:right;">
11
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
3.819485
</td>
<td style="text-align:right;">
0.6740911
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
3.74e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M6593_GSE5099_DAY3_VS_DAY7_MCSF_TREATED_MACROPHAGE_DN
</td>
<td style="text-align:right;">
120
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
1.610435
</td>
<td style="text-align:right;">
0.2880756
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
0e+00
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
5.32e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M561_chr19q13
</td>
<td style="text-align:right;">
527
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
-1.280877
</td>
<td style="text-align:right;">
0.2318304
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
1e-07
</td>
<td style="text-align:left;">
Down
</td>
<td style="text-align:right;">
7.15e-05
</td>
</tr>
<tr>
<td style="text-align:left;">
M5384_GSE3982_MEMORY_CD4_TCELL_VS_BCELL_UP
</td>
<td style="text-align:right;">
143
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
1.522163
</td>
<td style="text-align:right;">
0.2774059
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
1e-07
</td>
<td style="text-align:left;">
Up
</td>
<td style="text-align:right;">
8.31e-05
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="http://gabrielhoffman.github.io" class="external-link">Gabriel Hoffman</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
