<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Perform gene set analysis on the result of a pre-computed test statistic.  Test whether statistics in a gene set are larger/smaller than statistics not in the set."><title>Gene set analysis using pre-computed test statistic — zenithPR_gsa • zenith</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Gene set analysis using pre-computed test statistic — zenithPR_gsa"><meta property="og:description" content="Perform gene set analysis on the result of a pre-computed test statistic.  Test whether statistics in a gene set are larger/smaller than statistics not in the set."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">zenith</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/zenith.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/loading_genesets.html">Loading gene sets</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/DiseaseNeuroGenomics/zenith/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Gene set analysis using pre-computed test statistic</h1>
      <small class="dont-index">Source: <a href="https://github.com/DiseaseNeuroGenomics/zenith/blob/HEAD/R/zenithPR_gsa.R" class="external-link"><code>R/zenithPR_gsa.R</code></a></small>
      <div class="d-none name"><code>zenithPR_gsa.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Perform gene set analysis on the result of a pre-computed test statistic.  Test whether statistics in a gene set are larger/smaller than statistics not in the set.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">zenithPR_gsa</span><span class="op">(</span></span>
<span>  <span class="va">statistics</span>,</span>
<span>  <span class="va">ids</span>,</span>
<span>  <span class="va">geneSets</span>,</span>
<span>  use.ranks <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  n_genes_min <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  progressbar <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  inter.gene.cor <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  coef.name <span class="op">=</span> <span class="st">"zenithPR"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>statistics</dt>
<dd><p>pre-computed test statistics</p></dd>


<dt>ids</dt>
<dd><p>name of gene for each entry in <code>statistics</code></p></dd>


<dt>geneSets</dt>
<dd><p><code>GeneSetCollection</code></p></dd>


<dt>use.ranks</dt>
<dd><p>do a rank-based test <code>TRUE</code> or a parametric test <code>FALSE</code>? default: FALSE</p></dd>


<dt>n_genes_min</dt>
<dd><p>minumum number of genes in a geneset</p></dd>


<dt>progressbar</dt>
<dd><p>if TRUE, show progress bar</p></dd>


<dt>inter.gene.cor</dt>
<dd><p>correlation of test statistics with in gene set</p></dd>


<dt>coef.name</dt>
<dd><p>name of column to store test statistic</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<ul><li><p><code>NGenes</code>: number of genes in this set</p></li>
<li><p><code>Correlation</code>: mean correlation between expression of genes in this set</p></li>
<li><p><code>delta</code>: difference in mean t-statistic for genes in this set compared to genes not in this set</p></li>
<li><p><code>se</code>: standard error of <code>delta</code></p></li>
<li><p><code>p.less</code>: p-value for hypothesis test of <code>H0: delta &lt; 0</code></p></li>
<li><p><code>p.greater</code>: p-value for hypothesis test of <code>H0: delta &gt; 0</code></p></li>
<li><p><code>PValue</code>:  p-value for hypothesis test <code>H0: delta != 0</code></p></li>
<li><p><code>Direction</code>: direction of effect based on sign(delta)</p></li>
<li><p><code>FDR</code>: false discovery rate based on Benjamini-Hochberg method in <code>p.adjust</code></p></li>
<li><p><code>coef.name</code>: name for pre-computed test statistics. Default: <code>zenithPR</code></p></li>
</ul></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This is the same as <code><a href="zenith_gsa-methods.html">zenith_gsa()</a></code>, but uses pre-computed test statistics.  Note that <code>zenithPR_gsa()</code> may give slightly different results for small samples sizes, if <code>zenithPR_gsa()</code> is fed t-statistics instead of z-statistics.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="zenith_gsa-methods.html">zenith_gsa()</a></code>, <code><a href="https://rdrr.io/pkg/limma/man/camera.html" class="external-link">limma::cameraPR()</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># Load packages</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">edgeR</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioconductor.org/packages/variancePartition" class="external-link">variancePartition</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://www.creal.cat/jrgonzalez/software.htm" class="external-link">tweeDEseqCountData</a></span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Load RNA-seq data from LCL's</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">pickrell</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">geneCounts</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html" class="external-link">exprs</a></span><span class="op">(</span><span class="va">pickrell.eset</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">df_metadata</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html" class="external-link">pData</a></span><span class="op">(</span><span class="va">pickrell.eset</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Filter genes</span></span></span>
<span class="r-in"><span><span class="co"># Note this is low coverage data, so just use as code example</span></span></span>
<span class="r-in"><span><span class="va">dsgn</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">gender</span>, <span class="va">df_metadata</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">keep</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></span><span class="op">(</span><span class="va">geneCounts</span>, <span class="va">dsgn</span>, min.count<span class="op">=</span><span class="fl">5</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Compute library size normalization</span></span></span>
<span class="r-in"><span><span class="va">dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">geneCounts</span><span class="op">[</span><span class="va">keep</span>,<span class="op">]</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">dge</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">dge</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Estimate precision weights using voom</span></span></span>
<span class="r-in"><span><span class="va">vobj</span> <span class="op">=</span> <span class="fu"><a href="http://DiseaseNeurogenomics.github.io/variancePartition/reference/voomWithDreamWeights.html" class="external-link">voomWithDreamWeights</a></span><span class="op">(</span><span class="va">dge</span>, <span class="op">~</span> <span class="va">gender</span>, <span class="va">df_metadata</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Apply dream analysis</span></span></span>
<span class="r-in"><span><span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="http://DiseaseNeurogenomics.github.io/variancePartition/reference/dream-method.html" class="external-link">dream</a></span><span class="op">(</span><span class="va">vobj</span>, <span class="op">~</span> <span class="va">gender</span>, <span class="va">df_metadata</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">fit</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Load Hallmark genes from MSigDB</span></span></span>
<span class="r-in"><span><span class="co"># use gene 'SYMBOL', or 'ENSEMBL' id</span></span></span>
<span class="r-in"><span><span class="co"># use get_GeneOntology() to load Gene Ontology</span></span></span>
<span class="r-in"><span><span class="va">gs</span> <span class="op">=</span> <span class="fu"><a href="get_MSigDB.html">get_MSigDB</a></span><span class="op">(</span><span class="st">"H"</span>, to<span class="op">=</span><span class="st">"ENSEMBL"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using cached version from 2024-03-08 18:44:35</span>
<span class="r-in"><span>   </span></span>
<span class="r-in"><span><span class="co"># Run zenithPR analysis with a test statistic for each gene</span></span></span>
<span class="r-in"><span><span class="va">tab</span> <span class="op">=</span> <span class="fu">topTable</span><span class="op">(</span><span class="va">fit</span>, coef<span class="op">=</span><span class="st">'gendermale'</span>, number<span class="op">=</span><span class="cn">Inf</span><span class="op">)</span></span></span>
<span class="r-in"><span>  </span></span>
<span class="r-in"><span><span class="va">res.gsa</span> <span class="op">=</span> <span class="fu">zenithPR_gsa</span><span class="op">(</span><span class="va">tab</span><span class="op">$</span><span class="va">t</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">tab</span><span class="op">)</span>, <span class="va">gs</span><span class="op">)</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="http://gabrielhoffman.github.io" class="external-link">Gabriel Hoffman</a>.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer></div>

  

  

  </body></html>

