<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Gene set analysis following differential expression with dream — zenith • zenith</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Gene set analysis following differential expression with dream — zenith" />
<meta property="og:description" content="Perform gene set analysis on the result of differential expression using linear (mixed) modeling with variancePartition::dream by considering the correlation between gene expression traits.  This package is a slight modification of limma::camera to 1) be compatible with dream, and 2) allow identification of gene sets with log fold changes with mixed sign." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">zenith</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/geuvadis.html">Zenith gene set testing after dream analysis</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/GabrielHoffman/zenith/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Gene set analysis following differential expression with dream</h1>
    <small class="dont-index">Source: <a href='https://github.com/GabrielHoffman/zenith/blob/master/R/zenith.R'><code>R/zenith.R</code></a></small>
    <div class="hidden name"><code>zenith.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Perform gene set analysis on the result of differential expression using linear (mixed) modeling with <code><a href='https://rdrr.io/pkg/variancePartition/man/dream-method.html'>variancePartition::dream</a></code> by considering the correlation between gene expression traits.  This package is a slight modification of <code><a href='https://rdrr.io/pkg/limma/man/camera.html'>limma::camera</a></code> to 1) be compatible with dream, and 2) allow identification of gene sets with log fold changes with mixed sign.</p>
    </div>

    <pre class="usage"><span class='fu'>zenith</span><span class='op'>(</span>
  <span class='va'>fit</span>,
  <span class='va'>coef</span>,
  <span class='va'>index</span>,
  use.ranks <span class='op'>=</span> <span class='cn'>FALSE</span>,
  allow.neg.cor <span class='op'>=</span> <span class='cn'>FALSE</span>,
  progressbar <span class='op'>=</span> <span class='cn'>TRUE</span>,
  inter.gene.cor <span class='op'>=</span> <span class='cn'>NA</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>fit</th>
      <td><p>result of differential expression with dream</p></td>
    </tr>
    <tr>
      <th>coef</th>
      <td><p>coefficient to test using <code>topTable(fit, coef)</code></p></td>
    </tr>
    <tr>
      <th>index</th>
      <td><p>an index vector or a list of index vectors.  Can be any vector such that <code>fit[index,]</code> selects the rows corresponding to the test set.  The list can be made using <code>ids2indices</code>.</p></td>
    </tr>
    <tr>
      <th>use.ranks</th>
      <td><p>do a rank-based test (<code>TRUE</code>) or a parametric test ('FALSE')?</p></td>
    </tr>
    <tr>
      <th>allow.neg.cor</th>
      <td><p>should reduced variance inflation factors be allowed for negative correlations?</p></td>
    </tr>
    <tr>
      <th>progressbar</th>
      <td><p>if TRUE, show progress bar</p></td>
    </tr>
    <tr>
      <th>inter.gene.cor</th>
      <td><p>if NA, estimate correlation from data.  Otherwise, use specified value</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    
<ul>
<li><p><code>NGenes</code>: number of genes in this set</p></li>
<li><p><code>Correlation</code>: mean correlation between expression of genes in this set</p></li>
<li><p><code>delta</code>: difference in mean t-statistic for genes in this set compared to genes not in this set</p></li>
<li><p><code>se</code>: standard error of <code>delta</code></p></li>
<li><p><code>p.less</code>: p-value for hypothesis test of <code>H0: delta &lt; 0</code></p></li>
<li><p><code>p.greater</code>: p-value for hypothesis test of <code>H0: delta &gt; 0</code></p></li>
<li><p><code>PValue</code>:  p-value for hypothesis test <code>H0: delta != 0</code></p></li>
<li><p><code>Direction</code>: direction of effect based on sign(delta)</p></li>
<li><p><code>FDR</code>: false discovery rate based on Benjamini-Hochberg method in <code>p.adjust</code></p></li>
</ul>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p><code>zenith</code> gives the same results as <code>camera(..., inter.gene.cor=NA)</code> which estimates the correlation with each gene set.</p>
<p>For differential expression with dream using linear (mixed) models see Hoffman and Roussos (2020).  For the original camera gene set test see Wu and Smyth (2012).</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Hoffman GE, Roussos P (2020).
&#8220;dream: Powerful differential expression analysis for repeated measures designs.&#8221;
<em>Bioinformatics</em>.
doi: <a href='https://doi.org/10.1093/bioinformatics/btaa687'>10.1093/bioinformatics/btaa687</a>
.  Wu D, Smyth GK (2012).
&#8220;Camera: a competitive gene set test accounting for inter-gene correlation.&#8221;
<em>Nucleic acids research</em>, <b>40</b>(17), e133.
doi: <a href='https://doi.org/10.1093/nar/gks461'>10.1093/nar/gks461</a>
.</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='kw'><a href='https://rdrr.io/r/base/library.html'>library</a></span><span class='op'>(</span><span class='va'><a href='http://bioconductor.org/packages/variancePartition'>variancePartition</a></span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Loading required package: ggplot2</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: limma</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘limma’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:BiocGenerics’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    plotMA</span></div><div class='output co'>#&gt; <span class='message'>Loading required package: BiocParallel</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Attaching package: ‘variancePartition’</span></div><div class='output co'>#&gt; <span class='message'>The following object is masked from ‘package:limma’:</span>
#&gt; <span class='message'></span>
#&gt; <span class='message'>    classifyTestsF</span></div><div class='input'>
<span class='co'># simulate meta-data</span>
<span class='va'>info</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/data.frame.html'>data.frame</a></span><span class='op'>(</span>Age<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>20</span>, <span class='fl'>31</span>, <span class='fl'>52</span>, <span class='fl'>35</span>, <span class='fl'>43</span>, <span class='fl'>45</span><span class='op'>)</span>,Group<span class='op'>=</span><span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>0</span>,<span class='fl'>0</span>,<span class='fl'>0</span>,<span class='fl'>1</span>,<span class='fl'>1</span>,<span class='fl'>1</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># simulate expression data</span>
<span class='va'>y</span> <span class='op'>&lt;-</span> <span class='fu'><a href='https://rdrr.io/r/base/matrix.html'>matrix</a></span><span class='op'>(</span><span class='fu'><a href='https://rdrr.io/r/stats/Normal.html'>rnorm</a></span><span class='op'>(</span><span class='fl'>1000</span><span class='op'>*</span><span class='fl'>6</span><span class='op'>)</span>,<span class='fl'>1000</span>,<span class='fl'>6</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/paste.html'>paste0</a></span><span class='op'>(</span><span class='st'>"gene"</span>, <span class='fl'>1</span><span class='op'>:</span><span class='fl'>1000</span><span class='op'>)</span>
<span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>colnames</a></span><span class='op'>(</span><span class='va'>y</span><span class='op'>)</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/colnames.html'>rownames</a></span><span class='op'>(</span><span class='va'>info</span><span class='op'>)</span>

<span class='co'># First set of 20 genes are genuinely differentially expressed</span>
<span class='va'>index1</span> <span class='op'>&lt;-</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>20</span>
<span class='va'>y</span><span class='op'>[</span><span class='va'>index1</span>,<span class='fl'>4</span><span class='op'>:</span><span class='fl'>6</span><span class='op'>]</span> <span class='op'>&lt;-</span> <span class='va'>y</span><span class='op'>[</span><span class='va'>index1</span>,<span class='fl'>4</span><span class='op'>:</span><span class='fl'>6</span><span class='op'>]</span><span class='op'>+</span><span class='fl'>1</span>

<span class='co'># Second set of 20 genes are not DE</span>
<span class='va'>index2</span> <span class='op'>&lt;-</span> <span class='fl'>21</span><span class='op'>:</span><span class='fl'>40</span>

<span class='co'># perform differential expression analysis with dream</span>
<span class='va'>fit</span> <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/pkg/variancePartition/man/dream-method.html'>dream</a></span><span class='op'>(</span><span class='va'>y</span>, <span class='op'>~</span> <span class='va'>Age</span> <span class='op'>+</span> <span class='op'>(</span><span class='fl'>1</span><span class='op'>|</span><span class='va'>Group</span><span class='op'>)</span>, <span class='va'>info</span><span class='op'>)</span>
</div><div class='output co'>#&gt; <span class='message'>Dividing work into 50 chunks...</span></div><div class='output co'>#&gt; <span class='message'></span>
#&gt; <span class='message'>Total:16 s</span></div><div class='input'>
<span class='co'># perform gene set analysis testing Age</span>
<span class='va'>res</span> <span class='op'>=</span> <span class='fu'>zenith</span><span class='op'>(</span><span class='va'>fit</span>, <span class='st'>"Age"</span>, <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>set1<span class='op'>=</span><span class='va'>index1</span>,set2<span class='op'>=</span><span class='va'>index2</span><span class='op'>)</span> <span class='op'>)</span>

<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>res</span><span class='op'>)</span>
</div><div class='output co'>#&gt;      NGenes Correlation      delta        se    p.less p.greater    PValue
#&gt; set1     20  0.03288383  0.2820288 0.3003327 0.7994064 0.2005936 0.4011872
#&gt; set2     20  0.04809625 -0.1486478 0.3258152 0.3360195 0.6639805 0.6720390
#&gt;      Direction      FDR
#&gt; set1        Up 0.672039
#&gt; set2      Down 0.672039</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Gabriel Hoffman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


